### ISSUE TITLE
Missing translations in admin search dropdown

### ISSUE THREAD

**Issue by cpoint-eu** [sulu/sulu#6763]
| Q | A
| --- | ---
| Sulu Version | 2.5
| PHP Version | 8.1.6

# Actual Behavior

Admin search dropdown is not translated to user locale. 

# Possible Solutions

Pass locale query parameter to URL should solve this issue:

Current url:
/admin/search/indexes?flat=true

Possible solution:
/admin/search/indexes?flat=true&locale=de

**Comment by alexander-schranz** [comment]
I think the search is unlocalized by design and list content of all localizations, is that not the state in your case? The "result" will be shown in different "content locales". Currently there is no filter for a specific "locale" or any kind of filter. 

The locale which the user defined in its profile is the `System Locale` which only represnet the `Labels`, `Navigation`, ... Content is always in the Content Locale and the "System Locale" does never effect Content as it are 2 complete different localizations.

**Comment by cpoint-eu** [comment]
I don't mean the search result but the text translations in the dropdown menu:

All - translated correctly
Categories - always in english
Snippets - always in english
your.custom.label - untranslated (if you dont have en translations)

**Comment by alexander-schranz** [comment]
Okay understanding. Mostly this kind of issues screenshots helps. I will add one for if somebody want to tackle it:

<img width="312" alt="Bildschirmfoto 2022-08-03 um 11 05 14" src="https://user-images.githubusercontent.com/1698337/182569920-addc72af-b18f-4b64-9576-17b9b6f6b0d2.png">

I think the issue is somewhere here: https://github.com/sulu/sulu/blob/c4867cbe8b9f43e03fd2488c3156c0805c998858/src/Sulu/Bundle/SearchBundle/Search/Configuration/IndexConfigurationProvider.php#L32

Looking at the code I would think it was a bad idea that this was handled in `__construct` as there maybe the `translator` is in the false language and always fallback to `en`.

So something like:

```php
    private $indexConfigurations = [];

    public function getIndexConfigurations()
    {
        if (null === $this->indexConfigurations) {
            $this->indexConfigurations = // code from construct
        }
    
        return $this->indexConfigurations;
    }
    
    public function getIndexConfiguration($name)
    {
        $indexConfigurations = $this->getIndexConfigurations();

        if (!\array_key_exists($name, $indexConfigurations)) {
            return;
        }

        return $indexConfigurations[$name];
    }
```

Could maybe work.

@cpoint-eu do you want to give it a try if that would solve the problem and create a PR for 2.4.

**Comment by wachterjohannes** [comment]
fixed b #6765
